### ISSUE TITLE
[BUG] Odd things happen with more than 10 repeater fields

### ISSUE THREAD

**Issue by use** [bolt/bolt#6120]
Details
-------

 - Relevant Bolt Version: [ 3.2 ]
 - Install type: [ Composer install ]
 - PHP version: [ 7.0 ]
 - Used web server: [ Apache ]
 - For UX/UI issues: [ Chrome ]

Reproduction
------------
 1. **Bug summary**: 
    * When using a repeater field with more than 10 entries, odd things happen when trying to reorder them. Everything looks fine until you refresh the page and see that your data was not saved and ordered as expected. Rows of repeater fields entered by the user can be deleted by this bug.
 2. **Specifics**:
    * See below
 3. **Steps to reproduce**:
    1. Create a fresh bolt install (3.2)
    1. Edit contenttypes.yml to remove the `limit: 3` on `showcases` repeater (line 168)
    1. Create a new showcase and give it a required title like "test", then save
    1. Go to Repeater tab
    1. Add 11 repeater fields and give them some easily read titles like 1, 2, 3...
    1. Save
    1. Refresh the page, everything looks fine.
    1. On repeater field 11, click "Move Up"
    1. Save
    1. Refresh
    1. Repeater field 11 has now been deleted entirely
 4. **Expected result**: 
    * The repeater items would be correctly ordered and saved
 5. **Actual result**: 
    * The repeater items were not correctly ordered and are sometimes deleted

**Comment by GawainLynch** [comment]
> On repeater field 11, click "Move Up"

https://youtu.be/uMSV4OteqBE

**Comment by rossriley** [comment]
These only go up to 10.

**Comment by GawainLynch** [comment]
> These only go up to 10, it's one quieter.

But … 11

**Comment by rossriley** [comment]
But seriously, I thought we squashed this in #5581

**Comment by maetthu** [comment]
I've got the same issue, reproducible with following content type:

```YAML
testing:
  singular_slug: testing
  fields:
    title:
      type: text
    pages:
      type: repeater
      fields:
        body:
          type: html
    another:
      type: repeater
      fields:
        justsomefield:
          type: text
```

Adding more than 10 entries and saving the first time was fine, but reordering and then saving again leads to wrong order and removing content.

Attached screenshot shows the parameters sent when saving: Note the wrong index order (thought the record order is fine) and the duplicate index 10, resulting in an overwrite of the first value (I was moving "11" up two places to make it less loud ;)

![screenshot_20161203_204644](https://cloud.githubusercontent.com/assets/1193955/20861834/11b8226e-b99a-11e6-89dc-b1062fcd7dc9.png)

I was able to workaround the ordering issue by overwriting the _storage.request.save_ service and reindex the form values, though this does not fix the duplicate index issue.

```PHP
class Save extends \Bolt\Storage\ContentRequest\Save
{
    /**
     * @inheritDoc
     */
    public function action(array $formValues, array $contenttype, $id, $new, $returnTo, $editReferrer)
    {
        if($contenttype['slug'] === 'testing'){
            // reindex repeater field array bcz index does not correspond to correct order
            $formValues['another'] = array_values($formValues['another']);
        }

        return parent::action($formValues, $contenttype, $id, $new, $returnTo, $editReferrer);
    }
}
```

I don't think this is related to  #5581 because it seems to me that this is a client side issue only.

**Comment by maetthu** [comment]
I don't have a proper build environment right now, so I had to modify the compiled JS to test, but changing this line 

https://github.com/bolt/bolt/blob/6307ce400eb9e74b7f6140bc7a8e684ce38b50c4/app/src/js/widgets/fields/fieldRepeater.js#L193

from
```JS
re = new RegExp('^' + nameEsc + '\\\[(#|\\\d)\\\]');
```
to
```JS
re = new RegExp('^' + nameEsc + '\\\[(#|\\\d+)\\\]');
```

fixes the issue for me. (allow more than one digit in the regex).

**Comment by bobdenotter** [comment]
@maetthu Looks like a good fix to me! Thanks! 👍

**Comment by bobdenotter** [comment]
Fixed in #6136
